---
title: "T1D_MS"
author: "Sandeep Singh"
date: "14/01/2021"
output: 
   html_document:
    keep_md: true
always_allow_html: true
---

```{r global_options, include=FALSE}
knitr::opts_chunk$set(echo=FALSE, warning=FALSE, message=FALSE)
```


```{r}
# This document depends on the following packages:
library(rlang)
library(dplyr)
library(tibble)
library(readxl)
library(DT)
library(tidyr)
library(data.table)
library(ggplot2)
library(ggrepel)
library(forcats)
library(scales)
library(grid)
library(purrr) # to append multiple dataframes
library(plotly)
library(VennDiagram)
library(RColorBrewer)
library(png)
```

* raw_data.txt file was generated from previous researches on Type-1-diabetes and multiple sclerosis.  
* This file contains only 119 SNPs as common markers for T1D and MS. 

## File preparation
```{r}
T1D_MS_original_119 <- read.table("C:/SANDEEP/MAIN/CAREER/My_papers/PAPERS_in_PROGRESS/T1D_MS/T1D_MS_new/T1D.txt",
                                  header = F) 

colnames(T1D_MS_original_119) <- c("SNP")
# used to proivde column name

knitr::kable(T1D_MS_original_119)
```

* A total of 119 SNPs.
* This file was used to create ANNOVAR formatted file.

### File preparation by ANNOVAR
* Command used in ANNOVAR: 

convert2annovar.pl -format rsid T1D.txt -dbsnpfile humandb/hg19_snp138.txt > T1D_MS_formatted

* 119 SNPs records input; 110 SNPs records as output, 9 SNPs missing

```{r}
# duplicate SNPs due to HLA region were removed from ANNOVAR file
T1D_MS_original_110 <- read.table("C:/SANDEEP/MAIN/CAREER/My_papers/PAPERS_in_PROGRESS/T1D_MS/T1D_MS_new/T1D_MS_formatted.txt",
                                  header = F)

head(T1D_MS_original_110)
```

* Identify 9 missing SNPs

```{r}
# Chekcing difference of SNPs in T1D_MS_original_119 file and T1D_MS_original_110 file

setdiff(T1D_MS_original_119$SNP, T1D_MS_original_110$V6)
```

* missing SNP IDs were renamed from PUBMED

      rs6931717 > rs3095345
      
      rs7452756 > rs4516988
      
      rs12199773 > rs3130952
      
      rs6937967 >  rs3130534
      
      rs12178292 > rs2844558
      
      rs10456058 >  rs2734573
      
      rs7382662 >  rs3129305
      
      rs10807118 >  rs3130578
      
      rs10947377 > rs3130179
      
* Created a new file T1D_2.txt after chainging SNP ids
* Rerun in ANNOVAR and obtained results for all 119 SNPs 

```{r}
T1D_MS_new_119 <- read.table("file:///C:/SANDEEP/MAIN/CAREER/My_papers/PAPERS_in_PROGRESS/T1D_MS/T1D_MS_new/T1D_MS_formatted_2.txt", 
                             header = F)

head(T1D_MS_new_119)
```

# ANNOVAR ANALYSIS

## Gene based annotation

### Known Gene (UCSC)
```{r}
Known_T1D <- read.table("C:/SANDEEP/MAIN/CAREER/My_papers/PAPERS_in_PROGRESS/T1D_MS/T1D_MS_new/ANNOVAR_RESULTS/Gene_based/Known_gene_T1D.variant_function", header = F)
colnames(Known_T1D) <- c("Genomic region", "Gene", "Chromosome", "BP_1", "BP_2", "A1", "A2", "SNP") 

Known_T1D$BP_1 <- as.character(Known_T1D$BP_1)
head(Known_T1D)
```

### Ensembel Gene
```{r}
Ens_T1D <- read.table("C:/SANDEEP/MAIN/CAREER/My_papers/PAPERS_in_PROGRESS/T1D_MS/T1D_MS_new/ANNOVAR_RESULTS/Gene_based/Ens_gene_T1D.variant_function", header = F)
colnames(Ens_T1D) <- c("Genomic region", "Gene", "Chromosome", "BP_1", "BP_2", "A1", "A2", "SNP")

head(Ens_T1D)
```      

### Ref Gene (NCBI)
```{r}
Ref_T1D <- read.table("C:/SANDEEP/MAIN/CAREER/My_papers/PAPERS_in_PROGRESS/T1D_MS/T1D_MS_new/ANNOVAR_RESULTS/Gene_based/Ref_gene_T1D.variant_function", header = F)
colnames(Ref_T1D) <- c("Genomic_region", "Gene", "Chromosome", "BP_1", "BP_2", "A1", "A2", "SNP")

head(Ref_T1D)
```


* We will use known gene as reference to create table
```{r}
df <- Known_T1D %>% 
  select(`Genomic region`, SNP) %>%
  rename(Known_gene = 'Genomic region') %>% 
  left_join(Ens_T1D, by = 'SNP') %>% 
  select(Known_gene, 'Genomic region', SNP) %>% 
  rename(Ens_gene = 'Genomic region') %>% 
  left_join(Ref_T1D, by = "SNP") %>% 
  rename(Ref_gene = 'Genomic_region') %>% 
  select(SNP, Known_gene, Ens_gene, Ref_gene)

Genomic_region <- plot_ly() %>% 
  add_pie(data = count(df, Known_gene), labels = ~Known_gene, values = ~n, 
          textinfo = 'label+percent',
          name = "UCSC", domain = list(x = c(0, 0.4), y = c(0.5, 0.9))) %>% 
          add_pie(data = count(df, Ens_gene), labels = ~Ens_gene, values = ~n,           textinfo = 'label+percent',
          name = "Ensmbel", domain = list(x = c(0.6, 1), y = c(0.5, 0.9))) %>% 
          add_pie(data = count(df, Ref_gene), labels = ~Ref_gene, values = ~n,           textinfo = 'label+percent',
          name = "NCBI", domain = list(x = c(0.30, 0.70), y = c(0.2, 0.6))) %>% 
          layout(title = "Pie Charts showing genomic regions from different databases", showlegend = F,
                      xaxis = list(showgrid = FALSE, zeroline = FALSE, showticklabels = FALSE),
                      yaxis = list(showgrid = FALSE, zeroline = FALSE, showticklabels = FALSE))

Genomic_region
```

```{r}
# Donut plot
T1D_Ref <- Ref_T1D %>%
  group_by(Genomic_region) %>%
  summarize(count = n()) %>%
  plot_ly(labels =~Genomic_region, values = ~count) %>%
  add_pie(hole = 0.2,
          textinfo = 'label+percent',
          showlegend = FALSE)

T1D_Ref
```